Suppliers table

piole007 opened the pull request:
- Created the suppliers table in the database.
- Implemented GET and POST functionality for the suppliers API.
- The GET request fetches all suppliers from the database, and the POST request allows adding new suppliers.

Everything works so far :)

skovalevskis commented:
Let'swait for the https://github.com/piole007/horeca-analytics-mvp/pull/1 MR to be resolved first

skovalevskis commented:
The main idea looks good and it's ok for now to merge the PR, but in the future consider the following items:

- validating data from `req.body` object that came from the client side
- do not `select *`, provide each field separately to not send on the UI internal column of a table
- consider pagination (limit what amount of data can be returned back to the client side)

piole007 commented:
I have added pagination (10 pages per view)
Fixed select *, changed it to specific fields
Added input validation (the input should have supplier_name or contact_person and at least one way to contact supplier).

skovalevskis commented:
I think you need to rebase your branch with the latest commits from the `main` branch

```
$ git checkout suppliers_table
$ git fetch origin
$ git pull --rebase origin main
```

You'll need to resolve merge conflicts locally if there are any

piole007 commented:
Luckily no conflicts.
